schemas/v2/schema.yaml

```diff
@@ -1,97 +1,445 @@
-    script: String, optional # Select script defined in prerequisites, target can be referenced as `$script` in this task role.
-    extraContainerOptions:
-      shmMB: Integer, optional # Config the /dev/shm in a docker container, https://docs.docker.com/compose/compose-file/#shm_size.
-      infiniband: Boolean, optional # Use InfiniBand devices or not in a docker container.
-    resourcePerInstance:
-      cpu: Integer, required # CPU number, unit is CPU vcore
-      memoryMB: Integer, required # Memory number, unit is MB
-      gpu: Integer, required # GPU number, unit is GPU card
-      ports: # Optional, only for host network, port label is string in ^[a-zA-Z_][a-zA-Z0-9_]*$ format (valid C variable name).
-        <portLabel>: Integer, required, minimum number is 1 # Port number for the port label.
-    commands:
-      - String, required
-
-# To handle that a component may interact with different component differently, user is encouraged to place the codes handling such difference in the "deployments" field,
-# e.g., a job may get input data through wget, hdfs -dfs cp, copy, or just directly read from remote storage. This logic can be placed here.
-# In summary, the deployments field is responsible to make sure the job to run properly in a deployment specific runtime environment.
-# One could have many deployments, but only one deployment can be activated at runtime by specifying in "defaults". User can choose the deployment and specify in "defaults" at submission time.
-deployments:
-  - name: String, required
-    taskRoles:
-      <name>: String, required # Should be in taskRoles
-        preCommands:
-          - String, required # Execute before the taskRole's command
-        postCommands:
-          - String, required # Execute after the taskRole's command
-
-
-defaults: # Optional, default cluster specific settings
-  virtualCluster: String, optional
-  deployment: String, optional # Should reference to deployment defined in deployments
-
-extras: # Optional, extra field, object, save any information that plugin may use
-  submitFrom: String, optional
-  hivedscheduler: # Optional
-    jobPriorityClass: String, required
-    taskRoles:
-      <name>: String, required
-        gpuType/reservationId: String, required  # Only one allowed
-        affinityGroupName: String, optional
+openapi: 3.0.1
+info:
+  title: OpenPAI Protocol Specification
+  description: |
+    [OpenPAI](https://github.com/microsoft/pai) Protocol specification.
+  version: v2.0.0-alpha
+externalDocs:
+  description: Find out more about OpenPAI Protocol.
+  url: https://github.com/microsoft/openpai-protocol
+paths: {}
+components:
+  schemas:
+    JobProtocol:
+      type: object
+      description: Job protocol specification.
+      properties:
+        protocolVersion:
+          oneOf:
+            - type: number
+            - type: string
+          description: Protocol version, current version is 2.
+          example: 2
```

Comment:
it's not language specific

allowing both makes it easier to write config, `version: 2` is parsed as int in yaml, so you don't need to write `version: !!str 2` or `version: "2"`

Reply:
I agree with convenience, however, when I read current definition, I am not sure which of below are allowed (though they may be the same):
- `2` or `"2"` (definitely allowed)
- `2.0` or `"2.0"` (is it allowed? not sure)
- `2.0.0` (more confused) 

my point is to clarify the format of our versioning, e.g., we use **how many digits** to represent

Reply:
semantic versioning `major.minor.patch` is a good choice, but we used the version `2` at the very beginning, and it can be any string, e.g. a commit hash for script version.

for `protocolVersion` field, I will change the schema to `enum: ["2", 2]`. we may change the version in next major change, let's discuss the future format in the future.

Reply:
agreed. I create issue #5 to track it.